### PR TITLE
Use GSL MPAS model repository

### PR DESCRIPTION
**Pull request by hu5970**
Switch the upstream repository for MPAS model to NOAA-GSL/MPAS-Model.
Switch the upstream repository for MPASSIT to LarissaReames-NOAA/MPASSIT.git
Those two are used for GSL real-time MPAS runs.

### CODE REVIEW

File: .gitmodules
Comment (guoqing-noaa) [inline .gitmodules:23]: Larissa's version does not work on Gaea
The module file for Hera and Jet are not standard.
I think the best way is to sync what we need fro Larissa's version to the rrfsx fork until Larissa merges this PR https://github.com/LarissaReames-NOAA/MPASSIT/pull/42

File: .gitmodules
Comment (guoqing-noaa) [inline .gitmodules:9]: Change the branch name to match what we want

File: .gitmodules
Comment (hu5970) [reply .gitmodules:23]: This is to match the version used in real-time MPAS runs on Jet. We will figure out how to manage MPASSIT repository after we can repeat realtime MPAS runs.

File: .gitmodules
Comment (hu5970) [reply .gitmodules]: Done

File: .gitmodules
Comment (guoqing-noaa) [reply .gitmodules:23]: @hu5970 We can make it to repeat what David has on Jet and at the same work for Hera/Gaea as well. 

@daviddowellNOAA @hu5970 Do we still use a precompiled mpassit.x on Jet?  If not, could you let me know what branch/hash you have been using for Jet realtime runs from Larrissa's repo? I think it is very possible to repeat Jet realtime runs without scarifying other machines. Thanks!

Comment (daviddowellNOAA) [reply .gitmodules:23]: For Jet real-time runs, we're using the most current version (hash number bd91387) of code in https://github.com/LarissaReames-NOAA/MPASSIT.

Comment (guoqing-noaa) [reply .gitmodules:23]: Thanks, @daviddowellNOAA!

@hu5970 I've rebased RRFSx/MPASSIT up to Larissa's latest commit # bd91387
https://github.com/RRFSx/MPASSIT/commits/master/
Could you go ahead to point sorc/MPASSIT to the above latest master branch?

@daviddowellNOAA My commits did not involve any source code changes, only unifying the MPASSIT modulefiles across all platforms to use spack-stack.v1.6.0.  It can compile on Jet. You are welcome to try that.
If you want to do a quick test, here is the precompiled executable:
`/lfs5/BMC/wrfruc/gge/rrfs-workflow/exec/mpassit.x`
and be sure to load the following module before running MPASSIT:
```
module use sorc/MPASSIT/modulefiles
module load build.jet.intel
```

**Comment by EricJames-NOAA** [comment]
@guoqing-noaa @hu5970 based on discussion with Ming yesterday, should we point to this version of MPASSIT so that we can manage it more easily?

https://github.com/NOAA-GSL/MPASSIT

I have an open PR to Larissa's code, but not sure how long it will take her to merge that.  Using NOAA-GSL will allow us more flexibility to update MPASSIT.  

I should also try to compile it on Gaea now that I have access to C6.

**Comment by guoqing-noaa** [comment]
> @guoqing-noaa @hu5970 based on discussion with Ming yesterday, should we point to this version of MPASSIT so that we can manage it more easily?
> 
> 
> 
> https://github.com/NOAA-GSL/MPASSIT
> 
> 
> 
> I have an open PR to Larissa's code, but not sure how long it will take her to merge that.  Using NOAA-GSL will allow us more flexibility to update MPASSIT.  
> 
> 
> 
> I should also try to compile it on Gaea now that I have access to C6.

I don't know when this fork was created. For now, we have to use the rrfsx fork which has my latest commits to enable its use on multiple platforms based on spackstack 1.6.0. We can use NOAA-GSL in a later PR.

If you mean your PR about making MPAASIT compile on Hera, I think that was already superseded by the update to 1.6.0. We can talk more when I am back in office the week after next. Thanks!

**Comment by hu5970** [comment]
@guoqing-noaa Eric just created the NOAA-GSL/MPASSIT as official MPASSIT repository for RRFS development. Our current target is to use workflow to run MPAS to repeat GSL MPAS real-time runs on JET. Other portability enhancement in rrfsx can be merged later. Thanks.

**Comment by hu5970** [comment]
@EricJames-NOAA Which hash we should use from:
https://github.com/NOAA-GSL/MPASSIT

Also, this one repeats the one used by MPAS real-time runs, right?

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA Which hash we should use from: https://github.com/NOAA-GSL/MPASSIT
> 
> Also, this one repeats the one used by MPAS real-time runs, right?

@hu5970 On Friday, David updated the "HRRRv5" realtime MPAS runs to use my recent PR to https://github.com/LarissaReames-NOAA/MPASSIT

Since I'm not sure when Larissa will have time to merge my PR, I can try to put that code into https://github.com/NOAA-GSL/MPASSIT so that we can point to it here

**Comment by hu5970** [comment]
I have changed the MPASSIT submodule to follow NOAA-GSL/MPASSIT repository. We can put all enhancement of MPASSIT to this repository.

**Comment by hu5970** [comment]
@MatthewPyle-NOAA If there are no other comments on this PR, could you merge this? Thanks, Ming

**Comment by MatthewPyle-NOAA** [comment]
@hu5970 I'll merge it today if nobody objects.

**Review by BenjaminBlake-NOAA — APPROVED** [review]
@MatthewPyle-NOAA @hu5970 It looks good to me but let's wait and see if anyone who is more familiar with the changes has any comments.

**Review by MatthewPyle-NOAA — APPROVED** [review]
Looks okay to me, but will wait a bit to merge....

**Pull request merged**
